format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/PathTrackingDelimitedInputFormat.java

```diff
@@ -88,12 +96,19 @@ public NullWritable getCurrentKey() {
                 throw new IOException(message + " Did you mean to use the 'text' format?");
               }
             }
+            if (!enableQuotesValue && delimitedString.contains("\"")) {
+              message += " Check if quoted values should be allowed.";
+            }
             throw new IOException(message + " Check that the schema contains the right number of fields.");
           }
 
           if (part.isEmpty()) {
             builder.set(fields.next().getName(), null);
           } else {
+            // if this part contains the original delimeter, remove the quotes
+            if (enableQuotesValue && part.contains(delimeter)) {
```

Comment:
somebody could choose to quote something that doesn't contain the delimiter, we still want to remove the enclosing quotes in that case.